src/otx/algorithms/detection/configs/base/data/semisl/semisl_is_eff_data_pipeline.py

```diff
@@ -0,0 +1,157 @@
+            )
+        ],
+        p=0.8,
+    ),
+    dict(type="RandomGrayscale", p=0.2),
+    dict(
+        type="RandomApply",
+        transform_cfgs=[
+            dict(
+                type="RandomGaussianBlur",
+                sigma_min=0.1,
+                sigma_max=2.0,
+            )
+        ],
+        p=0.5,
+    ),
+    dict(type="PILImageToNDArray", keys=["img"]),
+    dict(type="Normalize", **__img_norm_cfg),
+    dict(type="Pad", size_divisor=32),
+    dict(type="NDArrayToTensor", keys=["img", "img0"]),
+    dict(
+        type="RandomErasing",
+        p=0.7,
+        scale=[0.05, 0.2],
+        ratio=[0.3, 3.3],
+        value="random",
+    ),
+    dict(
+        type="RandomErasing",
+        p=0.5,
+        scale=[0.02, 0.2],
+        ratio=[0.10, 6.0],
+        value="random",
+    ),
+    dict(
+        type="RandomErasing",
+        p=0.3,
+        scale=[0.02, 0.2],
+        ratio=[0.05, 8.0],
+        value="random",
+    ),
+]
+
+train_pipeline = [
+    dict(type="LoadImageFromOTXDataset", enable_memcache=True),
+    dict(
+        type="LoadAnnotationFromOTXDataset",
+        domain="instance_segmentation",
+        with_bbox=True,
+        with_mask=True,
+        poly2mask=False,
+    ),
+    dict(type="Resize", img_scale=__img_size, keep_ratio=False),
+    dict(type="RandomFlip", flip_ratio=0.5),
+    dict(type="Normalize", **__img_norm_cfg),
+    dict(type="Pad", size_divisor=32),
+    dict(type="DefaultFormatBundle"),
+    dict(
+        type="Collect",
+        keys=["img", "gt_bboxes", "gt_labels", "gt_masks"],
+        meta_keys=[
+            "ori_filename",
+            "flip_direction",
+            "scale_factor",
+            "img_norm_cfg",
+            "gt_ann_ids",
+            "flip",
+            "ignored_labels",
+            "ori_shape",
+            "filename",
+            "img_shape",
+            "pad_shape",
+        ],
+    ),
+]
+
+unlabeled_pipeline = [
+    dict(type="LoadImageFromOTXDataset", enable_memcache=True),
+    *common_pipeline,
+    dict(
+        type="ToDataContainer",
+        fields=[
+            dict(key="img", stack=True),
+            dict(key="img0", stack=True),
+        ],
+    ),
+    dict(
+        type="Collect",
+        keys=[
+            "img",
+            "img0",
+        ],
+    ),
+]
+
+test_pipeline = [
+    dict(type="LoadImageFromOTXDataset", enable_memcache=True),
+    dict(
+        type="MultiScaleFlipAug",
+        img_scale=__img_size,
+        flip=False,
+        transforms=[
+            dict(type="Resize", keep_ratio=False),
+            dict(type="Normalize", **__img_norm_cfg),
+            dict(type="Pad", size_divisor=32),
+            dict(type="ImageToTensor", keys=["img"]),
+            dict(type="Collect", keys=["img"]),
+        ],
+    ),
+]
+
+data = dict(
+    samples_per_gpu=4,
+    workers_per_gpu=2,
+    train=dict(type="RepeatDataset", times=13, dataset=dict(type=__dataset_type, pipeline=train_pipeline)),
```

Comment:
During my experiments repeat dataset brings improvement over base algorithm since we use small amount of data for semi-sl

Reply:
Sure, I rerun trainings now with latest code and I will put here a comparison table.

Reply:
Below experiments result with latest code

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/kprokofi/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/kprokofi/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
.font5
	{color:black;
	font-size:11.0pt;
	font-weight:700;
	font-style:normal;
	text-decoration:none;
	font-family:Calibri, sans-serif;
	mso-font-charset:0;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:Calibri, sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
.xl65
	{font-size:12.0pt;
	font-weight:700;}
-->
</style>
</head>

<body link="#0563C1" vlink="#954F72">



  | Pascal-Tiny mAP/F1 | Sea thrash | Cityscapes 8.5% | WGISD
-- | -- | -- | -- | --
Semi-SL + RD | 24.19 / 26.76 | 35.08 / 49.46 | 39.47 / 56.26 | 39.69 / 45.11
Semi-SL | 24.51 / 26.03 | 30.89 / 45.08 | 37.7 / 54.63 | 21.4 / 27.75



</body>

</html>

Reply:
As you can see RepeatDataset brings visible improvements especially for small dataset WGISD and also for larger datasets such as Sea Garbage and  Cityscapes

Reply:
Maybe we should also compare elapsed time. And do you think it's okay to use repeat dataset for large dataset such as COCO?

Reply:
Yeah, it is required more training time. For example for Sea Garbage dataset:
with RD: 12:18:20
w/o: 2:56:47
But the perfomance is also increased by 5% mAP

We have few options here, I guess
1) We can enable some additional parameter that will control enabling or disabling repeat dataset and change dataset config in Config Manager based on this parameter
2) We can enable automatic enabling or disabling repeat dataset for Semi-SL only based on some threshold (number of images) and also configure  in Config Manager
3) Leave Repeat Dataset for now enabled and wait for the future feedback if there will be some concerns about training time
4) Remove RD, but decrease Semi-SL perfomance by some margin

Reply:
@jaegukhyun what do you think?

Reply:
However I am not sure about threshold, I can't find proper number of images, RD brings improvements even for larger datasets. 
I stick to making switcher for RD as separate parameter
Since Geti target are small datasets - let's keep it "on" by default.

Reply:
I prefer dynamic repeat dataset. For example 
`times for repeat dataset = func(# of trained images + # of unlabeled images)`
func will return 1 ~ 13 integer.
However it requires more engineering, so I think it should be next step. Let's leave it as it is for this time

Reply:
I also have a similar concern with @jaegukhyun. If the user uses the COCO dataset (120K) images, then the total number of images will be 120K * 13. However, it could be handled at a later `AdaptiveRepeatDataHook` PR. 

For now, it only works at multi-class classification. (https://github.com/openvinotoolkit/training_extensions/pull/2428)

Reply:
I am not sure that 120k COCO dataset is valid example for our Semi-SL scenario. 120k images are already enough to obtain good fully supervised model. I will highlight it in documentation, that it can be switched off when user have sufficient number of data
But I agree we can do something like in classification